src/Peachpie.CodeAnalysis/FlowAnalysis/ExpressionAnalysis.cs

```diff
@@ -164,12 +164,26 @@ public override T VisitStaticStatement(BoundStaticVariableStatement x)
                 // analyse initializer
                 Accept(v.InitialValue);
 
-                State.SetLessThanLongMax(local, (v.InitialValue.ConstantValue.HasValue && v.InitialValue.ConstantValue.Value is long && (long)v.InitialValue.ConstantValue.Value < long.MaxValue));
+                if (v.InitialValue.ConstantValue.IsInteger(out long val))
+                {
+                    if (val < long.MaxValue)
+                        State.SetLessThanLongMax(local, true);
+
+                    if (val > long.MinValue)
```

Comment:
else ?

Reply:
Good point, fixed in e8f7755671f70fa15bd3f069454a58a54ffb6d0a.